lgn-provers/src/provers/v1/query/euclid_prover.rs

```diff
@@ -279,44 +226,54 @@ impl StorageQueryProver for EuclidQueryProver
         histogram!("zkmr_worker_proving_latency", "proof_type" => proof_type).record(time);
 
         debug!(
-            "single path leaf size in kB: {}",
+            "chunk-aggregation size in kB: {}",
             proof.len() / 1024
         );
 
         Ok(proof)
     }
 
-    fn prove_single_path_branch(
+    fn prove_non_existence(
         &self,
-        input: SinglePathBranchInput,
-        child_proof: Vec<u8>,
+        input: NonExistenceInput,
         pis: &DynamicCircuitPis,
     ) -> anyhow::Result<Vec<u8>>
     {
-        debug!("Proving single path branch");
+        debug!("Proving non-existence");
 
         let now = std::time::Instant::now();
 
-        let circuit_input = CircuitInput::new_single_path(
-            SubProof::new_child_proof(
-                child_proof,
-                input.child_position,
-            )?,
-            input.left_child_info,
-            input.right_child_info,
-            input.node_info,
-            input.is_rows_tree_node,
+        let primary_column_id = input.column_ids[0];
+        let secondary_column_id = input.column_ids[1];
+        let rest_column_ids = input.column_ids[2..].to_vec();
```

Comment:
Could you use constant there so you can use the same constant in the DQ side when creating the inputs ? 
Arbitrary indices are just waiting to be misused... :D

Reply:
Or even better if you can create a struct of stg that explicit them out instead of having to extract them manually on the worker side.